### DISCUSSION TITLE
Accessing parameter units in reports?

### DISCUSSION THREAD

**Discussion by miggland** [inventree/InvenTree#3526] [Q&A] [answered]
I'm testing out reports - or labels to be more specific.

I can successfully use the following to get an output:
```
<p>Params:<br />
{% for k,v in parameters.items %}
 {{ k }}: {{ v }} <br />
{% endfor %}
</p>
<p>
{{ parameters }}
</p>
```
which looks like this:
![image](https://user-images.githubusercontent.com/1314853/184381825-978e275e-73d1-46d5-9292-ccb518bb3533.png)

What I'm missing is the parameters unit, in this case 'mm' - any way to easily access this?

**Comment by SchrodingersGat** [comment, marked as answer]
I think something like the following should work:

```html
{% for param in part.parameters %}
{{ param.template.name }} : {{ param.data }} {{ param.template.units }}
{% endfor %}
```

Note here that you're using `part.parameters` to directly access the parameters, not the top-level `parameters` object which returns the simplified dict object.

**Reply by matmair** [reply]
@SchrodingersGat that might need to be included in the docs.

**Reply by miggland** [reply]
I can't get that to work, I get an error:
```
{
    "error": "TypeError",
    "error_class": "<class 'TypeError'>",
    "detail": "'RelatedManager' object is not iterable",
    "path": "/api/label/part/2/print/",
    "status_code": 500
}
```
This appears for parts and for stock items, whether the part has parameters or not.. Adding `{% if part.parameters %}` around doesn't help either.

**Reply by SchrodingersGat** [reply]
Ah, try `{% for param in part.parameters.all %}` ?

**Reply by miggland** [reply]
OK, that works:
```
{% for param in part.parameters.all %}
  {{ param.template.name }} : {{ param.data }} {{ param.template.units }}<br />
{% endfor %}
```
Definitely something which should be in the docs :)

Wouldn't it make sense to include the unit in the string output which is in parameters? I would assume including the unit would be the more common use case, and any calculation or so requiring the raw data less common - that case would then use the syntax above?
